chronos/handles.nim

```diff
@@ -48,8 +48,18 @@ proc setSockOpt*(socket: AsyncFD, level, optname, optval: int): bool =
                 sizeof(value).SockLen) < 0'i32:
     result = false
 
+proc setSockOpt*(socket: AsyncFD, level, optname: int, value: pointer,
+                 valuelen: int): bool =
+  ## `setsockopt()` for custom options (pointer and length).
+  ## Returns ``true`` on success, ``false`` on error.
+  result = true
+  if setsockopt(SocketHandle(socket), cint(level), cint(optname), value,
+                Socklen(valuelen)) < 0'i32:
+    result = false
+
 proc getSockOpt*(socket: AsyncFD, level, optname: int, value: var int): bool =
   ## `getsockopt()` for integer options.
+  ## Returns ``true`` on success, ``false`` on error.
```

Comment:
@arnetheduck sorry, but this change are not related to AsyncStreams, so if you so do not like it, i can remove it, but currently i wish to avoid Option[T] or Result[T] changes.